gammapy/background/reflected.py

```diff
@@ -198,36 +181,58 @@ def setup(self):
             self._angle + Angle("360deg") - self._min_ang - self.min_distance_input
         )
 
-        # Distance image
-        self._distance_image = _compute_distance_image(self.reference_map)
+        if self._min_ang < 0:
+            print("ISSUE self._min_ang=", self._min_ang)
+        if self.min_distance_input < 0:
+            print("ISSUE self.min_distance_input=", self.min_distance_input)
 
     def find_regions(self):
         """Find reflected regions."""
         curr_angle = self._angle + self._min_ang + self.min_distance_input
         reflected_regions = []
+        geom = self.reference_map.geom
+        _run_exclusion_mask = None
+        _pixel_excluded = False
+        if not self.exclusion_mask is None:
+            _run_exclusion_mask = self.exclusion_mask.reproject(self.reference_map.geom)
+            _run_exclusion_mask.data[np.where(np.isnan(_run_exclusion_mask.data))] = 1
+            _mask_array = np.where(_run_exclusion_mask.data < 0.99)
+            _excluded_coords = geom.get_coord().apply_mask(_mask_array).to_coordsys('CEL')
+            if len(_excluded_coords[0]) > 0 and len(_excluded_coords[1]) > 0 :
+                _excluded_pixcoords = PixCoord(*SkyCoord(*_excluded_coords, unit='deg').to_pixel(geom.wcs))
+                _pixel_excluded = True
+
         while curr_angle < self._max_angle:
-            test_pos = self._compute_xy(self._pix_center, self._offset, curr_angle)
-            test_reg = CirclePixelRegion(test_pos, self._pix_region.radius)
-            if not self._is_inside_exclusion(test_reg, self._distance_image):
-                refl_region = test_reg.to_sky(self.reference_map.geom.wcs)
-                log.debug("Placing reflected region\n{}".format(refl_region))
-                reflected_regions.append(refl_region)
+            _test_reg = self._create_rotated_reg(curr_angle)
+            if _pixel_excluded is False or not np.any(_test_reg.contains(_excluded_pixcoords)):
```

Comment:
Is this approach rapid enough? Does it impact the performances for exclusion masks with many excluded pixels?

How about testing if pixels inside the rotated region are excluded or not?

Reply:
Can you precise me how to quantify this and what is the threshold of 'rapid enough'?
I do not understand the second question